### ISSUE TITLE
Question: add extra request info to logger.error, logger.warn automatically

### ISSUE THREAD

**Issue by gingerlime** [rocketjob/rails_semantic_logger#54]
rails_semantic_logger already uses `append_info_to_payload` to let us configure what gets logged automatically with every request. This works really well.

However, in some cases, we manually log some errors, warnings, infos etc, i.e. `logger.warn("something went wrong here")`. The "problem" with this is that this particular log record is somehow detached from the request log message, so I can't easily trace it to the same request... The logs will look something like this

```
2018-02-16 10:57:03.495567 W [18:13144340] HomeController -- something went wrong here
2018-02-16 10:57:03.571882 D [18:13144340] (1.724ms) ActiveRecord::Base -- {:sql=>"BEGIN"}

...

2018-02-16 10:57:09.953765 I [18:13144340] (6.459s) HomeController -- Completed #index -- {:controller=>"HomeController", :action=>"index", :format=>"HTML", :method=>"GET", :path=>"/", :status=>200, :view_runtime=>6271.12, :db_runtime=>103.66, :geoip_country=>nil, :ip=>"1.2.3.4", :current_user=>"bob", :status_message=>"OK"}

```

I'm wondering what's the best way to add some request info into those manual logs ... I can probably wrap logger with some helper, but I'm not sure it's the best way. Any suggestions / best practice to achieve this in a clean and safe way?

**Comment by gingerlime** [comment]
Just a quick note: I can use `log_tags` for some basic request params (e.g. `uuid`, `remote_ip`), but some other things aren't so easily accessible. For example, if I want to append the current user id to every log request...

**Comment by reidmorrison** [comment]
In production we append the current user, IP address, etc. to all of our log entries using the log tags using the example from the docs:
~~~ruby
  config.log_tags = {
    request_id: :request_id,
    ip:         :remote_ip,
    user:       -> request { request.cookie_jar['login'] }
  }
~~~

**Comment by gingerlime** [comment]
Thanks @reidmorrison. That's pretty much what we ended up using... We use devise with redis sessions, so the current user isn't as easily accessible though (and digging into the internals felt weird and a bit brittle). So I was hoping for something that could allow accessing `current_user`. We ended up simply skipping the user. 

Sorry I should have posted an update. Thanks for taking the time to write back.
